client_restaurant/shared/src/commonMain/kotlin/data/gateway/LocalConfigurationGateway.kt

```diff
@@ -0,0 +1,52 @@
+package data.gateway
+
+import data.local.model.UserConfigurationCollection
+import domain.gateway.ILocalConfigurationGateway
+import io.realm.kotlin.Realm
+import io.realm.kotlin.ext.query
+
+class LocalConfigurationGateway(private val realm: Realm) : ILocalConfigurationGateway {
+
+    override suspend fun saveAccessToken(token: String) {
+        realm.write {
+            copyToRealm(UserConfigurationCollection().apply {
+                this.accessToken = token
+            })
+        }
```

Comment:
This will cause to save new record/document to Realm each time you call this function, and the first old records will stay in Realm and will never updated or even deleted.